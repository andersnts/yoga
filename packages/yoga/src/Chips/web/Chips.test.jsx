import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { MapPin, ChevronDown } from '@gympass/yoga-icons';

import { ThemeProvider, Chips } from '../..';

describe('<Chips />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips>Classes</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with disabled prop', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips disabled>Classes</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with more than one Chip', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips>Classes</Chips>
          <Chips>Gyms and studios</Chips>
          <Chips>Personal trainers</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with system props', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips
            fs="16"
            bg="#D8385E"
            color="#ffffff"
            width={150}
            elevation="small"
          >
            Classes
          </Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    describe('selected', () => {
      it('should match snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips selected>Classes</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with more than one Chip', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips>Classes</Chips>
            <Chips selected>Gyms and studios</Chips>
            <Chips selected>Personal trainers</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('counter', () => {
      it('should match snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8}>Classes</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with more than one Chip', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8}>Classes</Chips>
            <Chips counter={52} selected>
              Gyms and studios
            </Chips>
            <Chips counter={1732} selected>
              Personal trainers
            </Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('icon', () => {
      it('should match selected snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8} icon={[MapPin]}>
              Location
            </Chips>
            <Chips counter={8} icon={[MapPin, ChevronDown]}>
              Classes
            </Chips>
            <Chips icon={[MapPin, ChevronDown]}>Gyms and studios</Chips>
            <Chips icon={[MapPin]}>Personal trainers</Chips>
            <Chips counter={8} selected icon={[MapPin]}>
              Activities
            </Chips>
            <Chips counter={8} selected icon={[MapPin, ChevronDown]}>
              Language
            </Chips>
            <Chips selected icon={[MapPin, ChevronDown]}>
              Categories
            </Chips>
            <Chips selected icon={[MapPin]}>
              Availability
            </Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Events', () => {
    it('should toggle', () => {
      const toggleMock = jest.fn();

      const { getByText } = render(
        <ThemeProvider>
          <Chips onClick={toggleMock}>Classes</Chips>
        </ThemeProvider>,
      );

      fireEvent.click(getByText('Classes'));

      expect(toggleMock).toHaveBeenCalled();
    });

    it('should toggle using onToggle', () => {
      const toggleMock = jest.fn();

      const { getByText } = render(
        <ThemeProvider>
          <Chips onToggle={toggleMock}>Classes</Chips>
        </ThemeProvider>,
      );

      fireEvent.click(getByText('Classes'));

      expect(toggleMock).toHaveBeenCalled();
    });
  });
});
