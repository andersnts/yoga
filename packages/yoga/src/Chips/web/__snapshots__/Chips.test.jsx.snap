// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chips /> Snapshots counter should match snapshot 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots counter should match snapshot with more than one Chip 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  min-width: 16px;
  padding-right: 5px;
  padding-left: 5px;
  margin-left: 4px;
  color: #FFFFFF;
  font-size: 12px;
  line-height: 16px;
  font-weight: 500px;
  border-radius: 8px;
  background-color: #D8385E;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  background-color: #FFEEF2;
  color: #D8385E;
  border-color: transparent;
  font-weight: 500;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:hover:enabled {
  border-color: #D8385E;
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Gyms and studios
    </span>
    <span
      class="c3"
    >
      52
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Personal trainers
    </span>
    <span
      class="c3"
    >
      +999
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots icon should match selected snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  min-width: 16px;
  padding-right: 5px;
  padding-left: 5px;
  margin-left: 4px;
  color: #FFFFFF;
  font-size: 12px;
  line-height: 16px;
  font-weight: 500px;
  border-radius: 8px;
  background-color: #D8385E;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  background-color: #FFEEF2;
  color: #D8385E;
  border-color: transparent;
  font-weight: 500;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:hover:enabled {
  border-color: #D8385E;
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Location
    </span>
  </button>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Gyms and studios
    </span>
  </button>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Personal trainers
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Activities
    </span>
    <span
      class="c3"
    >
      8
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Language
    </span>
    <span
      class="c3"
    >
      8
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Categories
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Availability
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots selected should match snapshot 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  background-color: #FFEEF2;
  color: #D8385E;
  border-color: transparent;
  font-weight: 500;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #D8385E;
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots selected should match snapshot with more than one Chip 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  background-color: #FFEEF2;
  color: #D8385E;
  border-color: transparent;
  font-weight: 500;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:hover:enabled {
  border-color: #D8385E;
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Gyms and studios
    </span>
  </button>
  <button
    class="c2"
  >
    <span
      class="c1"
    >
      Personal trainers
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot with disabled prop 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

<div>
  <button
    class="c0"
    disabled=""
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot with more than one Chip 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

<div>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Gyms and studios
    </span>
  </button>
  <button
    class="c0"
  >
    <span
      class="c1"
    >
      Personal trainers
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot with system props 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px;
  border-radius: 8px;
  border-width: 1px;
  font-family: Rubik;
  font-size: 12px;
  line-height: 16px;
  border-color: #D7D7E0;
  background-color: #FFFFFF;
  color: #231B22;
  font-weight: 400;
  background-color: #D8385E;
  width: 150px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  color: #ffffff;
  font-size: 16px;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  color: #9898A6;
  border-color: #D7D7E0;
  cursor: not-allowed;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:focus:enabled,
.c0:active:enabled {
  border-color: rgba(215,215,224,0.75);
}

<div>
  <button
    class="c0"
    color="#ffffff"
    elevation="small"
    width="150"
  >
    <span
      class="c1"
    >
      Classes
    </span>
  </button>
</div>
`;
